mlabs/src/Mlabs/EfficientNFT/Token.hs

```diff
@@ -89,14 +88,22 @@ mkPolicy burnHash previousNft collectionNftP mintAct ctx =
             _ -> False
 
     -- Check if the old token is burnt and new is minted with correct name
-    checkMintAndBurn nft newPrice newOwner =
-      let oldName = mkTokenName nft
+    checkReMint nft newPrice newOwner =
+      let minted =
+            sortOn (\(_, _, amt) -> amt)
```

Comment:
I am not sure if that's the most efficient way of doing it.

Reply:
Tried to address this with hopefully more efficient implementation.